webapp-client/src/component/Navbar.js

```diff
@@ -17,20 +17,19 @@ export default class Navbar extends React.Component {
     this.onAuth2Init = this.onAuth2Init.bind(this);
   }
 
-  componentDidMount() {
-    window.gapi.load('auth2', this.gapiSetState);
+  async componentDidMount() {
+    await window.gapi.load('auth2', this.gapiSetState);
   }
 
-  gapiSetState() {
-    window.gapi.auth2
-      .init({
+  async gapiSetState() {
+    try {
+      await window.gapi.auth2.init({
         client_id: process.env.REACT_APP_CLIENT_ID,
-      })
-      .then(
-        this.onAuth2Init,
-        // Called if there is an init error
-        this.showErrorMessage
-      );
+      });
+      this.onAuth2Init();
+    } catch (error) {
+      this.showErrorMessage();
+    }
   }
 
   onAuth2Init() {
```

Comment:
IIUC the reason you moved this out of gapiSetState was because you needed a callback. Now that you're using await, you no longer need a separate function so you can just move this back into gapiSetState, right?